src/Http/Cookie/Cookie.php

```diff
@@ -0,0 +1,387 @@
+     * @return void
+     */
+    protected function validateName($name)
+    {
+        if (preg_match("/[=,; \t\r\n\013\014]/", $name)) {
+            throw new InvalidArgumentException(
+                sprintf('The cookie name `%s` contains invalid characters.', $name)
+            );
+        }
+
+        if (empty($name)) {
+            throw new InvalidArgumentException('The cookie name cannot be empty.');
+        }
+    }
+
+    /**
+     * Gets the raw cookie value
+     *
+     * @return string
+     */
+    public function getValue()
+    {
+        return $this->value;
+    }
+
+    /**
+     * Sets the raw cookie data
+     *
+     * @param string $value Value of the cookie to set
+     * @return $this
+     */
+    public function setValue($value)
+    {
+        $this->value = $value;
+
+        return $this;
+    }
+
+    /**
+     * Sets the path
+     *
+     * @param string $path Sets the path
+     * @return $this
+     */
+    public function setPath($path)
+    {
+        $this->path = $path;
+
+        return $this;
+    }
+
+    /**
+     * Sets the domain
+     *
+     * @param string $domain Domain to set
+     * @return $this
+     */
+    public function setDomain($domain)
+    {
+        $this->domain = $domain;
+
+        return $this;
+    }
+
+    /**
+     * Sets the expiration date
+     *
+     * @param \DateTimeInterface $dateTime Date time object
+     * @return $this
+     */
+    public function expiresAt(DateTimeInterface $dateTime)
+    {
+        $this->expiresAt = (int)$dateTime->format('U');
+
+        return $this;
+    }
+
+    /**
+     * Checks if a value exists in the cookie data
+     *
+     * @param string $path Path to check
+     * @return bool
+     */
+    public function check($path)
+    {
+        return Hash::check($this->value, $path);
+    }
+
+    /**
+     * Writes data to the cookie
+     *
+     * @param string $path Path to write to
+     * @param mixer $value Value to write
+     * @return $this
+     */
+    public function write($path, $value)
+    {
+        if (!$this->isExpanded) {
+            throw new RuntimeException('The Cookie data has not been expanded');
+        }
+
+        Hash::insert($this->value, $path, $value);
+
+        return $this;
+    }
+
+    /**
+     * Read data from the cookie
+     *
+     * @param string $path Path to read the data from
+     * @return mixed
+     */
+    public function read($path = null)
+    {
+        if (!$this->isExpanded) {
+            throw new \RuntimeException('The Cookie data has not been expanded');
+        }
+
+        if ($path === null) {
+            return $this->data;
+        }
+
+        return Hash::get($this->data, $path);
+    }
+
+    /**
+     * Encrypts the cookie value
+     *
+     * @param string $key Encryption key
+     * @return $this
+     */
+    public function encrypt($key)
+    {
+        $this->encryptionKey = $key;
+        $this->value = $this->_encrypt($this->value, 'aes', $key);
```

Comment:
Why do you always use `aes` here and then check for the cypher inside the function call ?